src/index.js

```diff
@@ -103,16 +79,16 @@ function passMetadata(s, context, metadata) {
 }
 
 module.exports = function(source, inputSourceMap) {
-  // Handle filenames (#106)
-  const webpackRemainingChain = loaderUtils
-    .getRemainingRequest(this)
-    .split("!");
-  const filename = webpackRemainingChain[webpackRemainingChain.length - 1];
+  const callback = this.async();
+  const filename = this.resourcePath;
```

Comment:
Oops 😛... I forgot to split this before opening the PR, if this should be a separate PR instead please let me know